src/grid.js

```diff
@@ -319,6 +320,19 @@ Grid.prototype.setSymmetry = function(symmetry) {
   this.sanitize();
 }
 Grid.prototype.setSize = function(width, height) {
+  // If the start and end are in the normal place, erase them and re-add them
+  // in the new normal places.
```

Comment:
Yeah, though it could be ambiguous whether you want the start to be "attached to the top" or "attached to the bottom".  I guess you could go by which side it's closer to.